lib/services/local/plugins.go

```diff
@@ -104,25 +107,56 @@ func (s *PluginsService) GetPlugin(ctx context.Context, name string, withSecrets
 
 // GetPlugins implements services.Plugins
 func (s *PluginsService) GetPlugins(ctx context.Context, withSecrets bool) ([]types.Plugin, error) {
-	startKey := backend.Key(pluginsPrefix)
-	result, err := s.backend.GetRange(ctx, startKey, backend.RangeEnd(startKey), backend.NoLimit)
+	const pageSize = 1000
+	var results []types.Plugin
+
+	var page []types.Plugin
+	var startKey string
+	var err error
+	for {
+		page, startKey, err = s.ListPlugins(ctx, pageSize, startKey, withSecrets)
+		if err != nil {
+			return nil, trace.Wrap(err)
+		}
+		results = append(results, page...)
+		if startKey == "" {
+			break
+		}
+	}
+
+	return results, nil
+}
+
+// ListPlugins returns a paginated list of plugin instances.
+// StartKey is a resource name, which is the suffix of its key.
+func (s *PluginsService) ListPlugins(ctx context.Context, limit int, startKey string, withSecrets bool) ([]types.Plugin, string, error) {
+	startKeyBytes := []byte(startKey)
+	if len(startKeyBytes) == 0 {
+		startKeyBytes = backend.Key(pluginsPrefix)
+	}
+	result, err := s.backend().GetRange(ctx, startKeyBytes, backend.RangeEnd(backend.Key(pluginsPrefix)), limit)
 	if err != nil {
-		return nil, trace.Wrap(err)
+		return nil, "", trace.Wrap(err)
 	}
 
 	plugins := make([]types.Plugin, 0, len(result.Items))
 	for _, item := range result.Items {
 		plugin, err := services.UnmarshalPlugin(item.Value, services.WithResourceID(item.ID), services.WithExpires(item.Expires))
 		if err != nil {
-			return nil, trace.Wrap(err)
+			return nil, "", trace.Wrap(err)
 		}
 		if !withSecrets {
 			plugin = plugin.WithoutSecrets().(types.Plugin)
 		}
 		plugins = append(plugins, plugin)
 	}
 
-	return plugins, nil
+	var nextKey string
+	if len(result.Items) == limit {
+		nextKey = backend.NextPaginationKey(plugins[len(plugins)-1])
```

Comment:
`startKey` is inclusive, so this will cause the item at the very end of the page to be seen twice.

Reply:
`NextPaginationKey` uses `nextKey` to increment by 1, so this should behave correctly? `TestListPlugins` should also verify this.

Reply:
You're right; in a lot of places we fetch `limit + 1` items when paginating just so the client doesn't have to do one extra API call if the page limit hits the last item exactly, but maybe we don't care for this specific case.

Reply:
quoting @rosstimothy: we should probably do pagination like we do in the other places, and that would be by fetching `limit + 1` items and using the name of the extra one as the `nextKey`; if we want to change how pagination should behave for the one-past-the-last page, we should do that consistently